docker.compose.image.swarm.yml

```diff
@@ -0,0 +1,108 @@
+version: "3.7"
+services:
+  db_gtm_1:
+    environment:
+      - PG_HOST=0.0.0.0
+      - PG_NODE=gtm_1
+      - PG_PORT=6666
+    image: pavouk0/postgres-xl:latest
+    command: docker-cmd-gtm
+    entrypoint: docker-entrypoint-gtm
+    volumes:
+      - db_gtm_1:/var/lib/postgresql
+    networks:
+      - db
+    #    healthcheck:
+    #      test: ["CMD", "docker-healthcheck-gtm"]
+    deploy:
+      mode: global
```

Comment:
You are correct. I misunderstood global mode. All I was trying to limit replicas to be safe but realise now --replicas=1 would do what I was expecint globa too. I have only tested it on a single node swarm currently but will on a multi node one as soon as it's working properly on one.

Reply:
`--replicas=1` would indeed do what you expect—however, it's the default, so in fact you don't need it. You don't actually need the constraints section at all; I just include it because I'm presuming you're running non-trivial clusters (i.e. greater than 1-node). However, if I were actually to use this, I'd likely change the constraints to constrain to `dbxl_coord_1` etc., so the containers would only ever be launched on nodes containing the data volume. Usually, this would constrain each container to be on a specific node, although of course if you had shared storage, it could also allow for safe failover of a specific Coordinator or Datanode still whilst assuming the 1-replica. Again, I think this is likely a bit out-of-scope, for this, especially as it would require you having a backend shared storage solution configured separately. Perfectly possible, though.